### PR TITLE
[IMP] reference/cmdline: corrects spelling errors, improves explicitness, adds references to missing cmd line options, groups opts into alphabetized categories for easier reference, adds separation of concerns between cmd line and config file, adds reference to config file options...

### PR DESCRIPTION
**Pull request by tesserB**
Description of the issue/feature this PR addresses:

This pull request aims to improve upon the structure, readability, and completeness of the "Command-line interface: odoo.py" documentation.

Current behavior before PR:

N/A

Desired behavior after PR is merged:

I hope to make it even easier for new Odoo developers to get started, or to save more seasoned developers the hassle of referring to command.py.
## 

I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr
Cody Kitterman

### CODE REVIEW

**Comment by tesserB** [comment]
@mart-e It's not ready to be merged, but was hoping for some input/guidance.
p.s. I apologize for crashing the party earlier.

**Comment by tesserB** [comment]
@fhe-odoo Would you mind checking these out for me, man, and letting me know if this is something Odoo would be willing to work with? I know it needs further editing, but I've asked others for input on the structural changes, and I haven't heard a peep. I understand some of that—"'everyone' already knows this stuff"—but I would like to finish these up, and move on to more significant improvements.

File: doc/reference/cmdline.rst [hunk 1]
```diff
@@ -6,282 +6,297 @@
...
-.. _reference/cmdline/server:
+.. _reference/cmdline/env:
 
-Running the server
-==================
+Installing Development Environment
+==================================
 
-.. program:: odoo.py
+    .. note:: This section applies to those developing within
```
Comment (mart-e) [inline doc/reference/cmdline.rst]: why the indentation? this makes a quote

File: doc/reference/cmdline.rst [hunk 2]
```diff
@@ -6,282 +6,297 @@
...
+such as finding the appropriate remotes; and setup Odoo's dependencies, including Postgres.
 
-.. option:: -u <modules>, --update <modules>
 
-    comma-separated list of modules to update before running the server
-    (requires :option:`-d`).
+.. _reference/cmdline/scaffold:
 
-.. option:: --addons-path <directories>
+Creating a Module Skeleton
```
Comment (mart-e) [inline doc/reference/cmdline.rst]: scaffold is not something you will use very often, even when developing, no need to put it at the top of the page (more useful to get basics running commands)

File: doc/reference/cmdline.rst [hunk 3]
```diff
@@ -6,282 +6,297 @@
...
 
-.. option:: --workers <count>
+The scaffolding command helps developers simplify the process of creating
+new modules. It does so by generating the appropriate files and file structure
+needed by even the most basic of Odoo modules.
 
-    if ``count`` is not 0 (the default), enables multiprocessing and sets up
-    the specified number of HTTP workers (sub-processes processing HTTP
-    and RPC requests).
+    .. note:: If you have experience with Django and it's "django-admin startproject"
```
Comment (mart-e) [inline doc/reference/cmdline.rst]: probably more `startapp` than `startproject`

**Comment by mart-e** [comment]
Hi,

Thanks for the PR, I have added a few comments inline but a few general remarks:

- Could you try to reduce your contributions to minimal commits, so it's easier to navigate through the diff (I am not sure if you just moved blocks or changed the content too)
- this PR is for master, the [published documentation](https://www.odoo.com/documentation/) is the 9.0 (at least until v10 is released). Except if your changes refers to features only available in 10, you can submit for 9 (but there is conflicts)
- check the indentation of your blocks, not sure why you increased it on many sections but it makes everything looks like a quote

**Comment by mart-e** [comment]
I have already merged the obvious spelling errors at 68c6a51

File: doc/reference/cmdline.rst (hunk 1 above)
Comment (tesserB) [inline doc/reference/cmdline.rst]: Hah, where's that GIF of panda smashing the keyboard? (Something stupid I did at the last second, after a long day. I knew better too...) Will fix...

File: doc/reference/cmdline.rst (hunk 2 above)
Comment (tesserB) [inline doc/reference/cmdline.rst]: Small section and so I thought to 'just get it out of the way'... You're right though... Will move to the very bottom...

File: doc/reference/cmdline.rst (hunk 3 above)
Comment (tesserB) [inline doc/reference/cmdline.rst]: Of course...

**Comment by tesserB** [comment]
@mart-e 

> Could you try to reduce your contributions to minimal commits, so it's easier to navigate through the diff (I am not sure if you just moved blocks or changed the content too)

Definitely.

> this PR is for master, the published documentation is the 9.0 (at least until v10 is released). Except if your changes refers to features only available in 10, you can submit for 9 (but there is conflicts).

I'm still trying to work out what goes where, exactly. (It could be said to be "purely cosmetic", for instance...)

I should have it finished on Friday-Saturday morning.

**Pull request closed without merging**